File: backend/app/api/v1/schemas/algorithm.py
from pydantic import BaseModel
from typing import List, Dict, Any


class AlgorithmParam(BaseModel):
    name: str
    type: str
    default: Any
    min: Any = None
    max: Any = None
    description: str = ""


class AlgorithmMetadata(BaseModel):
    id: str
    name: str
    description: str
    version: str
    is_active: bool
    parameters: List[AlgorithmParam]

File: frontend/src/types/algorithm.ts
// Algorithm-related types
export interface AlgorithmParameter {
  name: string;
  type: 'integer' | 'float' | 'string' | 'boolean';
  default: string | number | boolean;
  min?: number;
  max?: number;
  description: string;
}

export interface Algorithm {
  id: string;
  name: string;
  description: string;
  version: string;
  is_active: boolean;
  parameters: AlgorithmParameter[];
}
